web/issue_sorting/issue_sorting.ex

```diff
@@ -0,0 +1,167 @@
+defmodule Tudo.IssueSorting do
+  @moduledoc"""
+  Provides an api for listing, searching and sorting issues stored in the
+  database
+  """
+  use Tudo.Web, :controller
+  alias Tudo.{Issue, Repo}
+  alias Rummage.Ecto
+  use Rummage.Phoenix.Controller
+
+  def get_issues(rummage_param) do
+    {query, rummage} = Issue
+      |> Ecto.rummage(rummage_param)
+
+    {Repo.all(query), rummage}
+  end
+
+  def get_issues(rummage_param, search_params) do
+    rummage_param
+    |> Map.put(
+        "search",
+        Map.merge(rummage_param["search"],
+          %{"repo_name" => %{"assoc" => [],
+                             "search_term" => search_params["repo_name"],
+                             "search_type" => "ilike"},
+                            }))
+    |> get_issues
+  end
+
+  def labels_to_search(search_params) do
+    search_params
+    |> Map.delete("repo_name")
+    |> Map.to_list
+    |> Enum.filter(fn {_name, bool} ->
+      bool == "true"
+    end)
+    |> Enum.map(fn {name, _bool} ->
+      name
+    end)
+  end
+
+  def get_issues_by_labels(rummage_params, search_params, labels) do
+      {_query, rummage} = Issue
+      |> Ecto.rummage(rummage_params)
+
+      [sort, direction] =
+        case rummage["sort"]["field"] do
+          nil ->
+            [:gh_updated_at, :desc]
+          field ->
+            field
+            |> String.split(".")
+            |> Enum.map(&(String.to_atom(&1)))
```

Comment:
```elixir
|> Enum.map(&String.to_atom/1)
```

Be careful with `String.to_atom` [we encountered problems when trying to convert untrusted values](https://github.com/LDMW/app/blob/e9755e0ff1da63df305d406d687c0d016422e0b6/web/views/homepage_view.ex#L51)

Reply:
Oo that's weird and super useful to know.

Do you think it matters in this case? I think the only way the string could contain a quote is if someone actively tried to break the form by fiddling with the html elements, etc. And in that case it would error on that one person's process, which I guess is the expected result. Happy to use the safe version if you think it's needed.

Reply:
It depends if you have control over what you are converting, I couldn't really see exactly what it was so I thought I'd mention it

Reply:
I think in this situation it's okay, as the things we are using `String.to_atom` on are consistent. 

They're generated by rummage from the search params from our front end, so unless there's a breaking change to rummage (which could break the rest of our code) it's consistent.